internal/git/gitsync_processor.go

```diff
@@ -44,7 +56,23 @@ type GitSyncProcessor struct {
 	clusterName string
 }
 
-func watchRepo(ctx context.Context, restConfig *rest.Config, repo *v1.RepositoryPath, namespace string) error {
+// stores the Patched files  and their data
+
+type PatchedResource struct {
+	Before map[string]string
+	After  map[string]string
+}
+
+type KubernetesResource struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Name      string `yaml:"name"`
+		Namespace string `yaml:"namespace"` // TODO : what if namespace is not present
+	} `yaml:"metadata"`
+}
```

Comment:
for consistency with other Kubernetes resources, I think we would probably want to define your inner struct as its own `Metadata` struct type

Reply:
ok noted